terraform/kinesis_firehose_delivery_streams.tf

```diff
@@ -0,0 +1,593 @@
+    policy      = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "firehose:PutRecord",
+                "firehose:PutRecordBatch"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "arn:aws:firehose:*:*:deliverystream/${var.environment}-dot-sdc-cvpep-wydot-*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "firehose:DescribeDeliveryStream",
+                "firehose:ListDeliveryStreams",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Action": [
+                "s3:AbortMultipartUpload",
+                "s3:GetBucketLocation",
+                "s3:GetObject",
+                "s3:ListBucket",
+                "s3:ListBucketMultipartUploads",
+                "s3:PutObject"
+            ],
+            "Resource": [
+                "${var.data_lake_bucket_arn}/*",
+                "${var.data_lake_bucket_arn}"
+            ]
+        }
+    ]
+}
+    EOF
+}
+
+resource "aws_iam_role" "firehose_role" {
+    assume_role_policy    = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "firehose.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+    EOF
+    description           = "WYDOT Kinesis firehose role"
+    force_detach_policies = false
+    max_session_duration  = 3600
+    name                  = "${var.environment}-dot-sdc-cvpep-wydot-kinesis-role"
+    path                  = "/"
+    tags                  = {
+        Environment = var.environment,
+        Project = var.data_providers[2]["project"],
+        Team = var.data_providers[2]["team"]
+    }
+}
+
+resource "aws_iam_role_policy_attachment" "firehose_attach_policy" {
+    policy_arn = aws_iam_policy.firehose_managed_policy.arn
+    role = aws_iam_role.firehose_role.name
+}
+
+resource "aws_iam_role_policy" "firehose_inline_policy_1" {
+    name   = "oneClick_firehose_delivery_role_1530016803472"
+    policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Action": [
+                "glue:GetTableVersions"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Action": [
+                "s3:AbortMultipartUpload",
+                "s3:GetBucketLocation",
+                "s3:GetObject",
+                "s3:ListBucket",
+                "s3:ListBucketMultipartUploads",
+                "s3:PutObject"
+            ],
+            "Resource": [
+                "${var.data_lake_bucket_arn}",
+                "${var.data_lake_bucket_arn}/*",
+                "arn:aws:s3:::%FIREHOSE_BUCKET_NAME%",
```

Comment:
I forget if it was you I had this conversation with... can these be removed?

Reply:
This is what's currently in Prod.  In order to not change anything in Prod I left it as-is.